Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -261,11 +268,12 @@ extension NSManagedObjectContext {
 
         try payload.messages.forEach { _ = try saveMessage(payload: $0, channelDTO: dto, syncOwnReactions: true, cache: cache) }
 
-        if dto.needsPreviewUpdate(payload) {
-            dto.previewMessage = preview(for: payload.channel.cid)
+        if dto.needsPreviewUpdate(payload), let newPreview = preview(for: payload.channel.cid) {
```

Comment:
If there is no newPreview shouldn't we update the previewMessage to empty?

Reply:
This has been changed. In this case, this is not needed actually. But in some cases, it is. (If we are in a mid-page for example). Unit tests were added to cover these scenarios 👍